lib/Doctrine/Persistence/Mapping/RuntimeReflectionService.php

```diff
@@ -35,14 +35,19 @@ public function __construct()
      */
     public function getParentClasses($class)
     {
-        if (! class_exists($class)) {
+        if (! (class_exists($class) || trait_exists($class)) ) {
             throw MappingException::nonExistingClass($class);
         }
 
-        $parents = class_parents($class);
+        $parents = class_parents($class) + class_uses($class);
```

Comment:
This introduces a violation of this method's signature: https://github.com/doctrine/persistence/blob/9f226731048d8efbd42e3b0079c6fc353f3a6b96/lib/Doctrine/Persistence/Mapping/ReflectionService.php#L23

Also, it's inconsistent with its name.